apps/ehr/src/telemed/state/appointment/appointment.queries.ts

```diff
@@ -582,29 +590,132 @@ export const useGetCreateExternalLabResources = ({
   });
 };
 
-export function useGetUnsolicitedResultsResources(input: {
-  requestType: UnsolicitedResultsRequestType.UNSOLICITED_RESULTS_ICON;
-}): UseQueryResult<GetUnsolicitedResultsResourcesForIcon | undefined, Error>;
-export function useGetUnsolicitedResultsResources(input: {
-  requestType: UnsolicitedResultsRequestType.GET_ALL_TASKS;
-}): UseQueryResult<GetUnsolicitedResultsResourcesForTable | undefined, Error>;
-export function useGetUnsolicitedResultsResources({
-  requestType,
-}: GetUnsolicitedResultsResourcesInput): UseQueryResult<GetUnsolicitedResultsResourcesOutput | undefined, Error> {
+export function useDisplayUnsolicitedResultsIcon(
+  input: GetUnsolicitedResultsResourcesForIconInput
+): UseQueryResult<GetUnsolicitedResultsResourcesForIcon | undefined, Error> {
   const apiClient = useOystehrAPIClient();
+  const { requestType } = input;
+
   return useQuery({
     queryKey: ['get unsolicited results resources', requestType],
 
     queryFn: async () => {
-      return apiClient?.getUnsolicitedResultsResources({ requestType });
+      const data = await apiClient?.getUnsolicitedResultsResources(input);
+      if (data && 'tasksAreReady' in data) {
+        return data;
+      }
+      return;
+    },
+
+    enabled: Boolean(apiClient && FEATURE_FLAGS.LAB_ORDERS_ENABLED),
+    staleTime: 1000 * 15, // 15 seconds
+  });
+}
+
+export function useGetUnsolicitedResultsResourcesForTable(
+  input: GetUnsolicitedResultsResourcesForTableInput
+): UseQueryResult<GetUnsolicitedResultsResourcesForTable | undefined, Error> {
+  const apiClient = useOystehrAPIClient();
+  const { requestType } = input;
+
+  return useQuery({
+    queryKey: ['get unsolicited results resources', requestType],
+
+    queryFn: async () => {
+      const data = await apiClient?.getUnsolicitedResultsResources(input);
+      if (data && 'unsolicitedResultRows' in data) {
+        return data;
+      }
+      return;
```

Comment:
should return any value except undefined; null is ok. Please fix it and other similar places in queryFns.